### PR TITLE
Fix QIT linting errors

### PR DESCRIPTION
**Pull request by coreymckrill**
### Changes proposed in this Pull Request:

Minor code updates across several files to fix linting errors found by QIT CLI. The biggest change is to how some $_POST inputs are handled and sanitized.

### How to test the changes in this Pull Request:

Basically there should be no visual changes from this. Check to make sure none of the admin UI for Accommodation Bookings shows any escaped HTML tags or other glitches and that the form fields work correctly.

1. Make sure WooCommerce Bookings is installed and activated along with this plugin
2. Add a new product
3. Under Product Data change the product type to Accommodation product
4. Check the boxes for Has Persons and Has Resources
5. Fill in some data for the product fields relevant to accommodations
6. On the Persons tab, check the box for Enable Person Types
7. Add two or three person types and fill in their data
8. On the Resources tab, add two or three resources and fill in their data
9. Publish the new product
10. When the screen has refreshed, make sure all the data you filled in is still present in the form fields
11. Change some of the person and resource data and update the product
12. When the screen has refreshed, make sure the changes you made have persisted

### Changelog entry

> Dev - Fix linting errors found by the Quality Insights Toolkit.

### CODE REVIEW

File: includes/admin/class-wc-accommodation-booking-admin-panels.php [hunk 1]
```diff
@@ -153,7 +153,7 @@ public function save_product_data( $post_id ) {
 		);
 
 		foreach ( $meta_to_save as $meta_key => $sanitize ) {
-			$value = ! empty( $_POST[ $meta_key ] ) ? $_POST[ $meta_key ] : '';
+			$value = sanitize_text_field( $_POST[ $meta_key ] ?? '' );
```
Comment (vedanshujain) [inline includes/admin/class-wc-accommodation-booking-admin-panels.php]: I think you will also want to add a `wp_unslash` like so:

```suggestion
			$value = sanitize_text_field( wp_unslash( $_POST[ $meta_key ] ?? '' ) );
```

Comment (vedanshujain) [reply includes/admin/class-wc-accommodation-booking-admin-panels.php]: Also, I am wondering if there should be a check to make sure that this booking does belong to the user (or that the user is an administrator).

**Review by vedanshujain — COMMENTED** [review]
Added some minor comments, looks good otherwise

File: includes/admin/class-wc-accommodation-booking-admin-panels.php (hunk 1 above)
Comment (coreymckrill) [reply includes/admin/class-wc-accommodation-booking-admin-panels.php]: There are nonce and permission checks [right before](https://github.com/woocommerce/woocommerce/blob/94599a14cf5eb6efa5c706d3dc9bf491a756dbf4/plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php#L268) the action is fired that this function is hooked to. That seems sufficient to me, but let me know if you disagree.

Comment (vedanshujain) [reply includes/admin/class-wc-accommodation-booking-admin-panels.php]: makes sense, that looks sufficient

**Review by vedanshujain — APPROVED** [review]
LGTM!

**Review by dkotter — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.2.1
* Theme: Storefront Version: 4.2.0
* WooCommerce - Version 7.7.0
* PHP: 8.0.22
* Web Server: Nginx
* Browser: Chrome - Version 111.0.5563.64 
* OS: macOS Monterey V 12.3.1
* Git Branch:  update/qit-linting-errors

</Details>



**Test Results -** 

The tests were performed as per the steps mentioned above, and the following observations were made:

1. All the steps mentioned in the testing steps were executed without any issues.
2. The changes proposed in the Pull Request did not introduce any visual changes or glitches in the admin UI for Accommodation Bookings.
3. The form fields worked correctly as expected.
4. All the data filled in the form fields were present after publishing the product.
5. The changes made to the person and resource data were persisted after updating the product.
6. Old (already) created product has the same data as set at the time of publishing product. 
7. Old products as well as new products are working as expected. 

**Functional Demo /  Screencast -**


https://github.com/woocommerce/woocommerce-accommodation-bookings/assets/6384299/ccfd7f14-00a8-467d-bbc9-c32ada9bc326




**Next Step-** Ready to ~~Code Review(Woo)~~ skipping woo review as already reviewed. Moving to UAT.

**Review by ankitguptaindia — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
**Regression+Smoke Test Report-** ✅ 
------------

**Testing Environment -** 


<Details>

* WordPress: 6.2.2
* Theme: Storefront Version: 4.2.0
* WooCommerce - Version 7.7.0
* PHP: 8.0.22 
* Web Server: Nginx
* Browser: Chrome - Version 113


</Details>


Tested with Archive File created via 

```php woorelease.phar build https://github.com/woocommerce/woocommerce-accommodation-bookings/tree/smoke-testing```

- Composer version - 2.3.5
- npm version - 6.14.14
- node version - 14.17.5

#### Please note that this plugin is tested with the build created by specified versions of Composer, Node, and NPM. 



**Status-** Ready to Merge 🚀

**Pull request merged**